File: Python/week 2/Palindrome.py
number = int(input("Enter a number to check if a palindrome or not: "))

original = number
rev = 0
while number > 0:
    rev = rev * 10 + number % 10
    number = number // 10

if original == rev:
    print("The number is a palindrome")
else:
    print("The number is not a palindrome")

File: C/week 7/Intermediate/palindrome.c
#include <stdio.h>

int main()
{
    int n, reverse = 0, remainder, original;
    printf ("Enter a number to reverse: ");
    scanf ("%d", &n);
    original = n;

    while (n != 0)
    {
        remainder = n % 10;
        reverse = reverse * 10 + remainder;
        n /= 10;
    }

    if (original == reverse)
    {
        printf ("%d is a palindrome", original);
    }
    else
    {
        printf ("%d is not a palindrome", original);
    }
    
    return 0;
}